spec/models/follow_request_spec.rb

```diff
@@ -9,7 +9,7 @@
     let(:target_account) { Fabricate(:account) }
 
     it 'calls Account#follow!, MergeWorker.perform_async, and #destroy!' do
-      expect(account).to        receive(:follow!).with(target_account, reblogs: true, notify: false, uri: follow_request.uri, languages: nil, bypass_limit: true)
+      expect(account).to        receive(:follow!).with(target_account, { reblogs: true, notify: false, uri: follow_request.uri, languages: nil, bypass_limit: true })
```

Comment:
Hm, that change is pretty surprising considering `Account#follow!` expects keyword arguments, not a hash.

Reply:
Yeah, I wasn't able to generate the failure at all locally here (ruby 3.2.1 on MacOS) so was sort of just guessing that the circle ci failures were from the same issue as the Tag one that I did see locally.

I suspect it might be possible to avoid this failure by using a double splat in the call to force it to be seen as kwargs ... but w/out being able to get the failure locally I'm not sure how productive that will be.

Relevant rspec-mocks thread - https://github.com/rspec/rspec-mocks/issues/1460 - with various other projects making similar changes.

Reply:
Interesting. According to that issue, this should be resolved in `rspec-mocks` 3.12.1. Since your PR is moving to `rspec-mocks` 3.12.3, I think you can drop this change.

Reply:
Maybe?! I'll remove it in a WIP commit and see what circle does with it.

Reply:
Yup -- looks like it was that rspec-mocks version difference providing the fix.

This original PR that had the failures - https://github.com/mastodon/mastodon/pull/23797/files - was moving to 3.11.2 on rspec-mocks. When I originally pushed I hadn't paid close attention to that version bump being relevant here, and had assumed I'd see the same circle failures.

Anyway, I've updated this PR to be JUST the version bump in Gemfile and the changes to search service spec (which are actually correct/relevant to change given method signature). I'm assuming this will pass on circle now and is ready for review.